artifactory/commands/transferfiles/transfer.go

```diff
@@ -68,108 +72,144 @@ func (tdc *TransferFilesCommand) Run() (err error) {
-		return errorutils.CheckError(err)
-	}
 
-	sourceStorageInfo, targetStorageInfo, err := tdc.getSourceAndTargetStorageInfo()
-	if err != nil {
+	if err = tdc.initStorageInfoManagers(); err != nil {
 		return err
 	}
 
-	sourceRepos, err := tdc.getLocalRepositories(sourceStorageInfo, tdc.sourceServerDetails)
+	sourceLocalRepos, err := tdc.getSourceLocalRepositories()
 	if err != nil {
 		return err
 	}
-
-	targetRepos, err := tdc.getLocalRepositories(targetStorageInfo, tdc.targetServerDetails)
+	targetAllLocalRepos, err := tdc.getAllTargetLocalRepositories()
 	if err != nil {
 		return err
 	}
 
-	// Set progress bar
-	tdc.progressbar, err = NewTransferProgressMng(int64(len(sourceRepos)))
+	// Handle interruptions
+	finishStopping, newPhase := tdc.handleStop(srcUpService)
+	defer finishStopping()
+
+	// Set progress bar with the length of the taget local and build info repositories
+	tdc.progressbar, err = NewTransferProgressMng(int64(len(targetAllLocalRepos)))
 	if err != nil {
 		return err
 	}
 
-	// Handle interruptions
-	shouldStop := false
-	var newPhase transferPhase
-	finishStopping := tdc.handleStop(&shouldStop, &newPhase, srcUpService)
-	defer finishStopping()
+	// Transfer local repositories
+	if err := tdc.transferRepos(sourceLocalRepos, targetAllLocalRepos, false, newPhase, srcUpService); err != nil {
+		return tdc.cleanup(err, sourceLocalRepos)
+	}
 
-	for repo, buildInfoRepo := range sourceRepos {
-		if shouldStop {
-			break
-		}
+	// Transfer build-info repositories
+	sourceLocalRepos, err = tdc.transferBuildInfoRepos(sourceLocalRepos, targetAllLocalRepos, newPhase, srcUpService)
 
-		if _, exists := targetRepos[repo]; !exists {
-			log.Error("repository '" + repo + "' does not exist in target. Skipping...")
+	// Close progressBar and create CSV errors summary file
+	return tdc.cleanup(err, sourceLocalRepos)
+}
+
+func (tdc *TransferFilesCommand) initStorageInfoManagers() error {
+	tdc.sourceStorageInfoManager = utils.NewStorageInfoManager(tdc.sourceServerDetails)
+	if err := tdc.sourceStorageInfoManager.CalculateStorageInfo(); err != nil {
+		return err
+	}
+	tdc.targetStorageInfoManager = utils.NewStorageInfoManager(tdc.targetServerDetails)
+	return tdc.targetStorageInfoManager.CalculateStorageInfo()
+}
+
+func (tdc *TransferFilesCommand) transferRepos(sourceRepos []string, targetRepos []string,
+	buildInfoRepo bool, newPhase *transferPhase, srcUpService *srcUserPluginService) error {
+	for _, repoKey := range sourceRepos {
+		if tdc.ShouldStop() {
+			return nil
+		}
+		if !buildInfoUtils.IsStringInSlice(repoKey, targetRepos) {
+			log.Error("repository '" + repoKey + "' does not exist in target. Skipping...")
 			continue
 		}
 
-		repoSummary, err := getRepoSummaryFromList(sourceStorageInfo.RepositoriesSummaryList, repo)
+		repoSummary, err := tdc.sourceStorageInfoManager.GetRepoSummary(repoKey)
 		if err != nil {
 			log.Error(err.Error() + ". Skipping...")
-			continue
+			return nil
```

Comment:
I think we should continue and not return nil. This will actually skip repo (as the above log message says). This can happen in case the repo was deleted from the source, before the transfer process reached it.